packages/cli/src/frameworks/nextjs/index.ts

```diff
@@ -89,6 +92,58 @@ export class NextJs implements Framework {
           break;
       }
     }
+
+    const nextJsDir = await detectPagesOrAppDir(path);
```

Comment:
Let's create a separate function for all of this logic, and export it from this file so tests can be written.

Could you also wrap the function in a try catch block so if an error is thrown it doesn't break installs. It doesn't need to do anything in the catch.